File: 1000/CF2254C1_Marenol_easy_version.java

import java.util.*;

public class CF2254C1_Marenol_easy_version {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        while (t-- > 0) {

            int n = sc.nextInt();
            String a = sc.next();
            String b = sc.next();

            int onesA = 0;
            int onesB = 0;

            int balanceA = 0;
            int balanceB = 0;

            for (int i = 0; i < n; i++) {

                if (a.charAt(i) == '1') {
                    onesA++;

                    if (i % 2 == 0)
                        balanceA++;
                    else
                        balanceA--;
                }

                if (b.charAt(i) == '1') {
                    onesB++;

                    if (i % 2 == 0)
                        balanceB++;
                    else
                        balanceB--;
                }
            }

            if (onesA == onesB && balanceA == balanceB)
                System.out.println("YES");
            else
                System.out.println("NO");
        }

        sc.close();
    }
}

File: 1000/CF2254C1_Marenol_easy_version.py
t = int(input())

for _ in range(t):
    n = int(input())
    a = input()
    b = input()

    ones_a = a.count('1')
    ones_b = b.count('1')

    balance_a = 0
    balance_b = 0

    for i in range(n):
        if a[i] == '1':
            if i % 2 == 0:
                balance_a += 1
            else:
                balance_a -= 1

        if b[i] == '1':
            if i % 2 == 0:
                balance_b += 1
            else:
                balance_b -= 1

    if ones_a == ones_b and balance_a == balance_b:
        print("YES")
    else:
        print("NO")